assets/app/scripts/services/buildConfigs.js

```diff
@@ -0,0 +1,29 @@
+'use strict';
+
+angular.module("openshiftConsole")
+  .factory('BuildConfigsService', function(DataService, $filter){
+    function BuildConfigsService() {}
+
+    BuildConfigsService.prototype.canBuild = function($scope) {
+      if (!$scope.buildConfig) {
+        return false;
+      }
+
+      if ($scope.buildConfig.metadata.deletionTimestamp) {
+        return false;
+      }
+
+      if ($scope.buildConfigBuildsInProgress &&
+          $filter('hashSize')($scope.buildConfigBuildsInProgress[$scope.buildConfig.metadata.name]) > 0) {
```

Comment:
I'd rather pass the build config and builds in progress map as parameters rather than passing `$scope`. We shouldn't really be passing `$scope` to services even though we do it in other places.